### PR TITLE
feat(model): add Vector struct + operators, for easy linear algebra

### PR DESCRIPTION
**Pull request by anarchuser**
Depends on #29 and #28

### CODE REVIEW

File: src/controller/controller.cpp [hunk 1]
```diff
@@ -9,7 +9,7 @@ void Controller::init() {
     static Uniform distY = Uniform(0.0, world.HEIGHT);
     static Uniform distE = Uniform(MIN_FOOD_SIZE, MAX_FOOD_SIZE);
 
-    for (int i = 0; i < 10; i++) world.addIndividual(new Fred(10 * i, 10 * i, 0, 40));
+    world.addIndividual(new Piercy(0, 0, 0, 0.2 * world.ENERGY));
```
Comment (xeruf) [inline src/controller/controller.cpp]: you should add a switch for that instead of simply replacing the code

File: src/controller/controller.cpp [hunk 2]
```diff
@@ -27,7 +27,7 @@ long Controller::run() {
 }
 
 bool Controller::simulate() {
-    terminalview.render(world);
+    //terminalview.render(world);
```
Comment (xeruf) [inline src/controller/controller.cpp]: same here, add a flag to go into different modes

File: src/model/model.h [hunk 3]
```diff
@@ -3,15 +3,53 @@
...
+    double operator / (const Point & p) const;      // Evaluates the distance between the two points
+
+    Point operator + (const Point & p) const;
+    Point operator + (const Vector & v) const;
+
+    Point operator - (const Point & p) const;
+    Point operator - (const Vector & v) const;
+
+    /* Operators to calculate the vector between two points */
+    Vector operator , (const Point & p) const;
```
Comment (xeruf) [inline src/model/model.h]: NO

File: src/model/object/entity/individual/fred/fred.h [hunk 4]
```diff
@@ -1,4 +1,4 @@
-#ifndef EVE_FRED_H
+#ifndef EVE_PIERCY_H
```
Comment (xeruf) [inline src/model/object/entity/individual/fred/fred.h:1]: what

**Review by xeruf — CHANGES_REQUESTED** [review]

File: src/model/model.h (hunk 3 above)
Comment (anarchuser) [reply src/model/model.h]: removed

File: src/controller/controller.cpp (hunk 1 above)
Comment (anarchuser) [reply src/controller/controller.cpp]: resolved in #28

File: src/controller/controller.cpp (hunk 2 above)
Comment (anarchuser) [reply src/controller/controller.cpp]: resolved in #28

File: src/model/object/entity/individual/fred/fred.h (hunk 4 above)
Comment (anarchuser) [reply src/model/object/entity/individual/fred/fred.h:1]: implemented in #28

**Pull request closed without merging**